File: examples/python-examples/SignedDistancePointToMesh.dox.py
from meshlib import mrmeshpy

mesh = mrmeshpy.loadMesh("mesh1.ctm")
point = mrmeshpy.Vector3f(1.5, 1, 0.5)

a = mrmeshpy.findSignedDistance(point, mesh)
print(f"Signed distance between meshes is {a.dist}")

File: examples/cpp-examples/SignedDistancePointToMesh.dox.cpp
#include "MRMesh/MRMeshLoad.h"
#include "MRMesh/MRMeshProject.h"
#include "MRMesh/MRMesh.h"
#include <iostream>

int main()
{
    auto mesh1Res = MR::MeshLoad::fromAnySupportedFormat( "mesh1.ctm" );
    if ( !mesh1Res.has_value() )
    {
        std::cerr << mesh1Res.error();
        return 1;
    }
    auto point = MR::Vector3f( 1.5f, 1.0f, 0.5f );

    auto dist = MR::findSignedDistance( point, *mesh1Res );
    if ( dist )
        std::cout << "Signed distance from point to mesh " << dist->dist << "\n";
    return 0;
}
